src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1116,6 +1116,13 @@ public SolutionState AddAnalyzerReference(ProjectId projectId, AnalyzerReference
                 throw new ArgumentNullException(nameof(analyzerReference));
             }
 
+            //PROTOTYPE
+            if (_projectIdToGeneratorDriverMap.TryGetValue(projectId, out var driver))
+            {
+                driver = driver.WithGenerators(analyzerReference.GetGenerators());
```

Comment:
Can we rename this to WithAdditionalGenerators? Otherwise this looks like it might be throwing away generators discovered in a previous step.